docs/pages/build-reference/e2e-tests.mdx

```diff
@@ -1,763 +1,166 @@
-      binaryPath: 'android/app/build/outputs/apk/release/app-release.apk',
-    },
-  },
-  devices: {
-    simulator: {
-      type: 'ios.simulator',
-      device: {
-        type: 'iPhone 14',
-      },
-    },
-    emulator: {
-      type: 'android.emulator',
-      device: {
-        avdName: 'pixel_4',
-      },
-    },
-  },
-  configurations: {
-    'ios.debug': {
-      device: 'simulator',
-      app: 'ios.debug',
-    },
-    'ios.release': {
-      device: 'simulator',
-      app: 'ios.release',
-    },
-    'android.debug': {
-      device: 'emulator',
-      app: 'android.debug',
-    },
-    'android.release': {
-      device: 'emulator',
-      app: 'android.release',
-    },
-  },
-};
-```
-
-</Collapsible>
-
-<Collapsible summary="eas-hooks/eas-build-on-success.sh">
-
-```sh eas-hooks/eas-build-on-success.sh
-#!/usr/bin/env bash
+<Step label="5">
+### Build your app and run E2E tests on EAS Build
 
-function cleanup()
-{
-  echo 'Cleaning up...'
-  if [[ "$EAS_BUILD_PLATFORM" == "android" ]]; then
-    # Kill emulator
-    adb emu kill &
-  fi
-}
+To execute a custom build using the `build-and-maestro-test` profile that will build your app and run the Maestro E2E tests afterward, run the following command:
 
-if [[ "$EAS_BUILD_PROFILE" != "test"* ]]; then
-  exit
-fi
-
-# Fail if anything errors
-set -eox pipefail
-# If this script exits, trap it first and clean up the emulator
-trap cleanup EXIT
-
-ANDROID_EMULATOR=pixel_4
-
-if [[ "$EAS_BUILD_PLATFORM" == "android" ]]; then
-  # Start emulator
-  $ANDROID_SDK_ROOT/emulator/emulator @$ANDROID_EMULATOR -no-audio -no-boot-anim -no-window -use-system-libs 2>&1 >/dev/null &
-
-  # Wait for emulator
-  max_retry=10
-  counter=0
-  until adb shell getprop sys.boot_completed; do
-    sleep 10
-    [[ counter -eq $max_retry ]] && echo "Failed to start the emulator!" && exit 1
-    counter=$((counter + 1))
-  done
-
-
-  # Execute Android tests
-  if [[ "$EAS_BUILD_PROFILE" == "test" ]]; then
-    detox test --configuration android.release
-  fi
-  if [[ "$EAS_BUILD_PROFILE" == "test_debug" ]]; then
-    detox test --configuration android.debug
-  fi
-else
-  # Execute iOS tests
-  if [[  "$EAS_BUILD_PROFILE" == "test" ]]; then
-    detox test --configuration ios.release
-  fi
-  if [[ "$EAS_BUILD_PROFILE" == "test_debug" ]]; then
-    detox test --configuration ios.debug
-  fi
-fi
-```
+<Terminal cmd={['$ npx eas-cli@latest build --profile build-and-maestro-test']} />
 
-</Collapsible>
+<ContentSpotlight
+  alt="Logs of successful Maestro tests"
+  src="/static/images/eas-build/tests/03-logs.png"
+  className="max-w-[480px]"
+/>
 
-<Collapsible summary="eas.json">
+Any Maestro artifacts including screenshots taken when flow fails that are saved at `/Users/expo/.maestro/tests` (the default destination) will be automatically uploaded as build artifacts. You can download them from the build page.
```

Comment:
```suggestion
When the flow fails, any Maestro artifacts are automatically uploaded as build artifacts. This includes screenshots saved at `/Users/expo/.maestro/tests` (the default destination). You can download them from the build page.
```

Let's start with the use case here which is when the flow is failing and improve this for clarity.